simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/HeartbeatScheduler.java

```diff
@@ -19,22 +21,39 @@
     private StationState stationState;
     private StationMessageSender stationMessageSender;
 
+    private AtomicInteger heartBeatInterval;
+    private AtomicLong timeOfLastHeartbeatSent;
+
     public HeartbeatScheduler(StationState stationState, StationMessageSender stationMessageSender) {
         this.stationState = stationState;
         this.stationMessageSender = stationMessageSender;
+        this.heartBeatInterval = new AtomicInteger();
+        this.timeOfLastHeartbeatSent = new AtomicLong(0);
     }
 
-    public void scheduleHeartbeat(int heartbeatInterval) {
+    public void updateHeartbeat(int heartbeatInterval) {
```

Comment:
Why is it called updateHeartbeat? This method schedules a task with the delay. Consider naming it as: scheduleHeartBeatTask